pages/dkp/kommander/2.2/dkp-upgrade/upgrade-kommander/index.md

```diff
@@ -14,34 +13,79 @@ This section describes how to upgrade your Kommander Management cluster and all
+    done < <(tar xfO "${AIRGAPPED_TAR_FILE}" "index.json" | grep -oP '(?<="io.containerd.image.name":").*?(?=",)')
+    ```
+
+Based on the network latency between the environment of script execution and the docker registry, it can take a while to upload all the images to your image registry.
 
 ## Upgrade Kommander
 
-Before running the following command, ensure that your `dkp` configuration **references the Kommander Management cluster**, otherwise it attempts to run the upgrade on the bootstrap cluster. You can do this by setting the `KUBECONFIG` environment variable [to the appropriate kubeconfig file's location][k8s-access-to-clusters].
+Before running the following command, ensure that your `dkp` configuration **references the Kommander Management cluster**, otherwise it attempts to run the upgrade on the bootstrap cluster. You can do this by setting the `KUBECONFIG` environment variable [to the appropriate kubeconfig file's location][k8s_access_to_clusters].
+
+<p class="message--note"><strong>NOTE:</strong> An alternative to initializing the KUBECONFIG environment variable as stated earlier is to use the <code>–kubeconfig=cluster_name.conf</code> flag. This ensures that Kommander upgrades on the workload cluster.</p>
 
-<p class="message--note"><strong>NOTE:</strong> An alternative to initializing the KUBECONFIG environment variable as stated earlier is to use the <code>–kubeconfig=cluster_name.conf</code> flag. This ensures that Kommander is upgraded on the workload cluster.</p>
+1.  Use the DKP CLI to upgrade Kommander and all the Platform Applications in the Management Cluster:
+
+    For air-gapped:
+
+    ```bash
+    dkp upgrade kommander --charts-bundle bundle.tar.gz
+    ```
 
-1.  Use the DKP CLI to upgrade Kommander and all the Platform Applications in the Management cluster:
+    For non air-gapped:
 
     ```bash
     dkp upgrade kommander
     ```
 
-    <A confirmation message appears once upgrade is complete.>
+    <--! A confirmation message appears once upgrade is complete.--! >
 
-1.  <Pending step: if confirmation message cannot be added on time, document a way to check the status of the upgrade in Management cluster.>
+1.  < --!Pending step: if confirmation message cannot be added on time, document a way to check the status of the upgrade in Management Cluster. --! >
 
-1.  Upgrade your [Workspaces and their Platform applications][upgrade-workspaces] on a per-Workspace basis, if your environment has additional Workspaces (with Managed and Attached clusters). Otherwise, proceed with the [Konvoy Upgrade][konvoy-upgrade].
+1.  Upgrade your additional [Workspaces][upgrade_workspaces] on a per-Workspace basis to upgrade the Platform Applications on other clusters than the Management Cluster. Do this only if your environment has additional Workspaces (with Managed and Attached clusters). Otherwise, proceed with the [Konvoy Upgrade][konvoy_upgrade]. < --! we need to reference this file once it is created-->
```

Comment:
> Do this only if your environment has additional Workspaces

I do not quite understand this sentence - the managed cluster belongs to it's own workspace and IIRC customers should not be using it for other clusters.

Reply:
Also, do we plan to add instructions on how to do the workspaces upgrade - i.e. which commands to call?

Reply:
@vespian With this sentence, I was trying to state that this is only necessary in a multi-cluster environment. But I agree that this only adds more confusion.  How about the following? 
```suggestion
1.  For Enterprise customers: Upgrade your additional [Workspaces][upgrade_workspaces] on a per-Workspace basis to upgrade the Platform Applications on other clusters than the Management Cluster. 
For Essential customers: Proceed with the [Konvoy Upgrade][konvoy_upgrade]. < --! we need to reference this file once it is created-->
```
And yes, we are working on a separate document where the Workspace Upgrade will live.

Reply:
> For Enterprise customers: Upgrade your additional [Workspaces][upgrade_workspaces] on a per-Workspace basis to upgrade the Platform Applications on other clusters than the Management Cluster. 

lgtm

> For Essential customers: Proceed with the [Konvoy Upgrade][konvoy_upgrade]. < --! we need to reference this file once it is created-->

So konvoy upgrade should happen after kommander upgrade?

Reply:
AFAIK, in a single-cluster experience, yes. / In a multi-cluster experience, all apps (platform, catalog) in non-management clusters must be upgraded first. That is why I would like to point to different documentation depending on the license model. (Essential customers can upgrade Konvoy next, but Enterprise customers have to follow other steps first)